### PR TITLE
Bluetooth: controller: Refactor ll c into states/roles files

### PR DESCRIPTION
**Pull request by cvinayak**
Collection of commit to refactor ll.c file.

---
This change is [Reviewable](https://reviewable.io/reviews/zephyrproject-rtos/zephyr/15)

### CODE REVIEW

**Review by Vudentz — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

**Comment by cvinayak** [comment]
@jhedberg @carlescufi @Vudentz Please do the needful!

File: subsys/bluetooth/controller/hci/hci.c [hunk 1]
```diff
@@ -952,17 +974,21 @@ static int controller_cmd_handle(u8_t ocf, struct net_buf *cmd,
 		le_rem_dev_from_wl(cmd, evt);
 		break;
 
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_STATE_CONN)
 	case BT_OCF(BT_HCI_OP_LE_CONN_UPDATE):
 		le_conn_update(cmd, evt);
 		break;
 
+#if defined(CONFIG_BLUETOOTH_CONTROLLER_ROLE_MASTER)
```
Comment (jhedberg) [inline subsys/bluetooth/controller/hci/hci.c]: Could you try to reorder the case statments so that each #if-#endif pair only appears once?

Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c]: Sure.

Lets see if this comment is lost!

Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c]: No this review/comment is not lost when doing git push -f.

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**